test/odbc/psqlodbc/smallmoney.cpp

```diff
@@ -0,0 +1,453 @@
+#include "psqlodbc_tests_common.h"
+
+const string TABLE_NAME = "master_dbo.smallmoney_table_odbc_test";
+const string COL1_NAME = "pk";
+const string COL2_NAME = "data";
+const string DATATYPE_NAME = "sys.smallmoney";
+const string VIEW_NAME = "master_dbo.smallmoney_view_odbc_test";
+const vector<pair<string, string>> TABLE_COLUMNS = {
+  {COL1_NAME, "INT PRIMARY KEY"},
+  {COL2_NAME, DATATYPE_NAME}
+};
+const int DOUBLE_BYTES_EXPECTED = 8;
+
+class PSQL_DataTypes_Smallmoney : public testing::Test {
+  void SetUp() override {
+    if (!Drivers::DriverExists(ServerType::PSQL)) {
+      GTEST_SKIP() << "PSQL Driver not present: skipping all tests for this fixture.";
+    }
+
+    OdbcHandler test_setup(Drivers::GetDriver(ServerType::PSQL));
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+
+  void TearDown() override {
+    if (!Drivers::DriverExists(ServerType::PSQL)) {
+      GTEST_SKIP() << "PSQL Driver not present: skipping all tests for this fixture.";
+    }
+
+    OdbcHandler test_teardown(Drivers::GetDriver(ServerType::PSQL));
+    test_teardown.ConnectAndExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+    test_teardown.CloseStmt();
+    test_teardown.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+// Helper to convert string to double
+double stringToDouble(const string &value) {
+  if (value == "NULL") {
+    return 0;
+  }
+  return std::stod(value);
+}
+
+vector<double> getExpectedResults_Smallmoney(vector<string> data) {
```

Comment:
I will move this function and `stringToDouble` to the `psql_odbc_tests_common` file in another PR.